### ISSUE TITLE
Wine sandbox setting creates wrong paths

### ISSUE THREAD

**Issue by FlyingWombat** [lutris/lutris#864]
Related to issue #744  
In Wine/Winesteam runner configuration:
"Create a sandbox for wine folders" checkbox
"Sandbox directory" field

tl;dr
----
browse button and field automatically change home directory to the shell expansion symbol '~'
but the field does not recognize this as an expansion symbol.

Issue
-------
Used browse button to select /home/$USER/Documents/My Games/lutris-desktop
\~/Documents/My Games/lutris-desktop  is what appeared in the field

Upon launching the game (see Note), lutris had changed the desktop integration folders to point to \~/Documents/My Games/lutris-desktop/My Documents
Unfortunately this is a broken link, since there is no '\~' directory in the wine prefix user directory.
Additionally, lutris created a /home/$USER/~/Documents/My Games/lutris-desktop/My Documents folder (as well as the other desktop integration directories (e.g. My Music))

The same thing happens if I change the \~ to $HOME, in the Sandbox Directory field.
So, the field is not doing any env variable expansion (which it probably shouldn't).

When I type in the field so it reads
/home/myuser/Documents/My Games/lutris-desktop
and save it, lutris automatically changes /home/myuser back to ~
But, this time it actually worked, and the symlinks were pointing to the correct spot.
So, I guess, when I enter the full path in the text field, it works,
but when I use the browse button to select the folder, it doesn't.

Let me know if you need any more info about this.

Note :
Lutris seems to create the symlinks to the wine sandbox directory upon loading a game, 
rather than just when creating a wine prefix.
So, if I manually make any links, or if there is existing content,
in the default lutris wine prefix, they are changed.
I could see the reason for always creating these links upon game start,
to make sure they are always there, but it makes more sense to me to set the links upon prefix creation.
Thankfully, it seems lutris is smart about renaming an exist My Documents folder,
to My Documents.winecfg
If I manually create the correct links, lutris will restore them to their incorrect versions upon game load.

Steps to reproduce: 
---------
1. Restore 'Sandbox Directory' to default in wine and winesteam settings.
2. Use *browse* button to select \$HOME/path/to/sandbox/dir and save settings
3. launch winesteam game (didn't try other runners)

some info:
System         : Arch Linux 4.16.8
Name           : lutris-git
Version         : 0.4.16.r9.baf76b87-1

**Comment by strycore** [comment]
Thanks for the report, it's now fixed in master
